src/coreComponents/constitutive/fluid/twophasefluid/TwoPhaseFluid.hpp

```diff
@@ -0,0 +1,353 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2016-2024 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2024 Total, S.A
+ * Copyright (c) 2018-2024 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2023-2024 Chevron
+ * Copyright (c) 2019-     GEOS/GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file TwoPhaseFluid.hpp
+ */
+
+#ifndef GEOS_CONSTITUTIVE_FLUID_TWOPHASEFLUID_TWOPHASEFLUID_HPP_
+#define GEOS_CONSTITUTIVE_FLUID_TWOPHASEFLUID_TWOPHASEFLUID_HPP_
+
+#include "common/DataLayouts.hpp"
+#include "functions/TableFunction.hpp"
+#include "constitutive/ConstitutiveBase.hpp"
+
+#include "constitutive/fluid/multifluid/Layouts.hpp"
+#include "constitutive/fluid/multifluid/MultiFluidUtils.hpp"
+
+#include "constitutive/ConstitutivePassThruHandler.hpp"
+
+
+namespace geos
+{
+namespace constitutive
+{
+
+class TwoPhaseFluid : public ConstitutiveBase
+{
+public:
+
+  TwoPhaseFluid( string const & name,
+                 Group * const parent );
+
+  virtual std::unique_ptr< ConstitutiveBase >
+  deliverClone( string const & name,
+                Group * const parent ) const override;
+
+  virtual void allocateConstitutiveData( dataRepository::Group & parent,
+                                         localIndex const numConstitutivePointsPerParentIndex ) override;
+
+  /**
+   * @name Static Factory Catalog members and functions
+   */
+  ///@{
+
+  /// string name to use for this class in the catalog
+  static constexpr auto m_catalogNameString = "TwoPhaseFluid";
+
+  /**
+   * @brief Static catalog string
+   * @return A string that is used to register/lookup this class in the registry
+   */
+  static std::string catalogName() { return m_catalogNameString; }
```

Comment:
```suggestion
  static string catalogName() { return "TwoPhaseFluid"; }
```
I would just hard-code this here instead of having the static member variable.